feat: Basic structure for UserServices and update on gitignore

Implycitt opened the pull request:
- Basic scaffold for the file UserService.cs
- Gitignore update

Some of the functions will have a few additions once we figure out what they will be returning exactly. 
Otherwise, they are mostly TODOs for until I get the controllers finished up and I'll come back to this once they are done.

liangricky7 reviewed (changes requested):
remember to have public class UserService, change DTO return types to Task<[dto name]> (read https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task?view=net-9.0) (tldr: it is the equivalent of JS's promise)

Implycitt commented:
The new push consists of about three parts:
- UserController.cs:
  - all functions to talk to the database by getting, updating, and deleting users based on their GUID.
- Updated DTOs to reflect the database structure 
  -  created a UserRole DTO to interact with the user role table
- fixed UserService.cs based on your previous comment.

liangricky7 reviewed (changes requested):
This controller is more what i'd expect out of services and not what a controller looks like; watch this: https://www.youtube.com/watch?v=cSnoAGHKVus

also, controller is missing a constructor which would be:
    private readonly Supabase.Client _supabase;

    public UserController(Supabase.Client supabase)
    {
        _supabase = supabase;
    }

Note that if u watched the video u may see ApplicationDbContext; replace all of that with Supabase stuffs to make it applicable here

liangricky7 reviewed (changes requested):
- DTOs do not need the [Table("user")] and other identifiers, only the model does
- I just realized u need to go make a user model (User.cs)
- ADDITIONALLY, for UserResponseDto, you would need stuff like [JsonPropertyName("[data_field_name]")] attributes above each thing to make the JSON deserializer work; reference the supabase for each data field name (i just learned this mb)
good work so far
